### PR TITLE
Add synchronization of commissioning proposals

### PR DESCRIPTION
**Pull request by stuartcampbell**
Also improved a couple of the method in `pass_service`

Now update the proposals with the cycle name when we sync a whole cycles worth of proposals.

### CODE REVIEW

File: src/nsls2api/services/proposal_service.py [hunk 1]
```diff
@@ -581,6 +581,30 @@ async def synchronize_proposal_from_pass(proposal_id: int) -> None:
...
+    proposal_list = await fetch_proposals_for_cycle(cycle_name)
+
+    logger.info(f"Found {len(proposal_list)} proposals for cycle {cycle_name}.")
+
+    for proposal_id in proposal_list:
+        # Add the cycle to the Proposal object
+
+        try:
+            proposal = await proposal_by_id(int(proposal_id))
+            await proposal.update(AddToSet({Proposal.cycles: cycle_name}))
```
Comment (padraic-shafer) [inline src/nsls2api/services/proposal_service.py:601]: Does this need to be an array -- to make it append to a list of cycles?

```suggestion
            await proposal.update(AddToSet({Proposal.cycles: [cycle_name]}))
```

I don't know beanie, thought it seemed to reference the mongodb operator of the same name...
https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/#value-to-add-is-an-array

File: src/nsls2api/services/pass_service.py [hunk 2]
```diff
@@ -121,11 +121,33 @@ async def get_saf_from_proposal(
...
+            for commissioning_proposal in pass_commissioning_proposals:
+                commissioning_proposal_list.append(PassProposal(**commissioning_proposal))
+    except ValidationError as error:
+        error_message = f"Error validating commissioning proposal data recevied from PASS for year {str(year)} at {facility} facility."
+        logger.error(error_message)
+        raise PassException(error_message) from error
+    except Exception as error:
+        error_message = f"Error retrieving commissioning proposal information from PASS for year {str(year)} at {facility} facility."
+        logger.exception(error_message)
+        raise PassException(error_message) from error
```
Comment (padraic-shafer) [inline src/nsls2api/services/pass_service.py:148]: This should probably be two separate try-catch blocks

1 for fetching proposals with PassException

1 within the for loop for validating each proposal.

Unless the entire call should fall if one bad proposal is returned?

File: src/nsls2api/services/pass_service.py [hunk 3]
```diff
@@ -211,7 +233,22 @@ async def get_proposals_allocated(
         raise PassException(error_message)
 
     url = f"{base_url}/Proposal/GetProposalsAllocated/{api_key}/{pass_facility}"
-    allocated_proposals = await _call_pass_webservice(url)
+
+    try:
```
Comment (padraic-shafer) [inline src/nsls2api/services/pass_service.py:237]: Same comment as above

**Review by padraic-shafer — COMMENTED** [review]
Do you want to merge as is, and consider the comments in a separate PR (if needed)?

File: src/nsls2api/services/proposal_service.py (hunk 1 above)
Comment (stuartcampbell) [reply src/nsls2api/services/proposal_service.py:601]: No it doesn't need to be an array in python

**Review by padraic-shafer — APPROVED** [review]
Will merge and raise issue to possibly revisit the try-catch blocks.

**Pull request merged**